### PR TITLE
Test: refine Executor Unit Test

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #4609

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/tests/gtest_window_executor.cpp [hunk 1]
```diff
@@ -438,7 +421,7 @@ try
                   .window(Lead2(concat(col("value"), concat(lit(Field(String("0"))), col("value"))), lit(Field(static_cast<UInt64>(1)))), {"order", false}, {"partition", false}, MockWindowFrame())
                   .build(context);
     result.emplace_back(toNullableVec<String>({"b0b", "c0c", "d0d", {}, "f0f", "g0g", "h0h", {}}));
-    executeWithConcurrency(request, result);
+    executeAndAssertColumnsEqual(request, result);
```
Comment (ywqzzy) [inline dbms/src/Flash/tests/gtest_window_executor.cpp:424]: How can we print the line number in gtest_xxx.cpp when test failed？

File: dbms/src/TestUtils/ExecutorTestUtils.cpp [hunk 2]
```diff
@@ -104,6 +104,38 @@ void ExecutorTest::executeInterpreter(const String & expected_string, const std:
     ASSERT_EQ(Poco::trim(expected_string), Poco::trim(fb.toString()));
 }
 
+void ExecutorTest::executeExecutor(
+    const std::shared_ptr<tipb::DAGRequest> & request,
+    std::function<void(const ColumnsWithTypeAndName &)> assert_func)
+{
+    WRAP_FOR_DIS_ENABLE_PLANNER_BEGIN
+    std::vector<size_t> concurrencies{1, 2, 10};
```
Comment (xzhangxian1008) [inline dbms/src/TestUtils/ExecutorTestUtils.cpp:112]: Maybe we should provide some  ways to set the concurrency policy, such as number of  concurrencies or degree of concurrency, by users.

File: dbms/src/Flash/tests/gtest_window_executor.cpp (hunk 1 above)
Comment (SeaRise) [reply dbms/src/Flash/tests/gtest_window_executor.cpp:424]: good point

Comment (SeaRise) [reply dbms/src/Flash/tests/gtest_window_executor.cpp:424]: now the fail msg will be
```
Value of: assert_func(executeStreams(request, concurrency))
  Actual: false (
Columns rows mismatch
expected_rows: 0
actual_rows: 1)
Expected: true
test info:
    file: /data1/huanghaisheng/git-repo/tiflash/dbms/src/Flash/tests/gtest_limit_executor.cpp
    line: 50
    test_case_name: ExecutorLimitTestRunner
    test_func_name: Limit
    enable_planner: false
    concurrency: 1
    block_size: 1
    dag_request: 
limit_1 | 1
 table_scan_0 | {<0, String>}
```

File: dbms/src/TestUtils/ExecutorTestUtils.cpp (hunk 2 above)
Comment (SeaRise) [reply dbms/src/TestUtils/ExecutorTestUtils.cpp:112]: I think they can use `executeStream(dagrequest, concurrency)` directly.
Specifying a specific degree of concurrency is a rare requirement.

**Review by ywqzzy — APPROVED** [review]

**Comment by xzhangxian1008** [comment]
lgtm

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**